app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayImageActivity.kt

```diff
@@ -1,28 +1,67 @@
 package com.github.HumanLearning2021.HumanLearningApp.view
 
+import android.content.Intent
 import android.os.Bundle
 import android.view.View
 import android.widget.TextView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
 import com.github.HumanLearning2021.HumanLearningApp.R
 import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseManagement
+import kotlinx.coroutines.launch
 
 class DisplayImageActivity : AppCompatActivity() {
+
+    private lateinit var picture: CategorizedPicture
+    private lateinit var datasetId: String
+    private val staticDBManagement = DummyDatabaseManagement.staticDummyDatabaseManagement
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_display_image)
 
-        val image: CategorizedPicture =
-            intent.getParcelableExtra("display_image_image")!!
-
-        findViewById<TextView>(R.id.display_image_viewCategory).text = image.category.name
-        image.displayOn(this, findViewById(R.id.display_image_viewImage))
+        picture =
+            intent.getParcelableExtra("single_picture")!!
+        datasetId = intent.getStringExtra("dataset_id")!!
+        findViewById<TextView>(R.id.display_image_viewCategory).text = picture.category.name
+        picture.displayOn(this, findViewById(R.id.display_image_viewImage))
     }
 
     fun removePicture(view: View) {
```

Comment:
For consistency it might be good to bind this to the button in the code instead of have it in the xml layout file, as in other parts of the code it is that way.